js/app/controllers/attendeecontroller.js

```diff
@@ -124,4 +143,28 @@ app.controller('AttendeeController', function($scope, AutoCompletionService) {
 		});
 		$scope.nameofattendee = '';
 	};
+
+	$scope.didUserSetupEmail = () => ($scope.$parent.emailAddress !== '');
+	$scope.isAttendeeListEmpty = () => ($scope.$parent.properties.attendee.length === 0);
+
+	$scope.isEMailInAttendeeList = (emailAddress) => {
+		console.log($scope.$parent.properties.attendee, $scope.mailtoHelper(emailAddress));
+
+		return !!$scope.$parent.properties.attendee.filter(
+			(elm) => elm.value === $scope.mailtoHelper(emailAddress)).length;
+	};
+
+	$scope.mailtoHelper = (emailAddress) => {
+		return 'MAILTO:' + emailAddress;
+	};
+
+	$scope.validateEMail = (emailAddress) => {
+		const helperInput = document.createElement('input');
+		helperInput.type = 'email';
+		helperInput.value = emailAddress;
+
+		return typeof helperInput.checkValidity === 'function' ?
+			helperInput.checkValidity() :
+			/^[a-zA-Z0-9.!#$%&'*+\/=?^_`{|}~-]+@[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?(?:\.[a-zA-Z0-9](?:[a-zA-Z0-9-]{0,61}[a-zA-Z0-9])?)*$/.test(emailAddress);
```

Comment:
Even if this regex looks good enough, I think validating emails is quite trickery. Is it done elsewhere in Nextcloud (at least to have consistency) ?

Reply:
Yes, validating email is pretty nasty, but that regex allows pretty much anything as long as there is an @ sign.
The source for this regex is https://www.w3.org/TR/html51/sec-forms.html#email-state-typeemail
So it's pretty much the same as `checkValidity`